libyul/AssemblyStack.cpp

```diff
@@ -135,13 +141,15 @@ void AssemblyStack::optimize(Object& _object)
 	for (auto& subNode: _object.subObjects)
 		if (auto subObject = dynamic_cast<Object*>(subNode.get()))
 			optimize(*subObject);
-	// TODO: Store this as setting - it should be the same as the flag passed to
-	// ::assemble(...)
-	bool optimizeStackAllocation = false;
-	OptimiserSuite::run(languageToDialect(m_language, m_evmVersion), *_object.code, *_object.analysisInfo, optimizeStackAllocation);
+	OptimiserSuite::run(
+		languageToDialect(m_language, m_evmVersion),
+		*_object.code,
+		*_object.analysisInfo,
+		m_optimiserSettings.optimizeStackAllocation
+	);
 }
 
-MachineAssemblyObject AssemblyStack::assemble(Machine _machine, bool _optimize) const
+MachineAssemblyObject AssemblyStack::assemble(Machine _machine) const
 {
```

Comment:
`test/tools/ossfuzz/strictasm_assembly_ossfuzz.cpp` needs to be adapted to invoke this API correctly.